File: sorting_algorithms/bubble-sort/bubbleSort.go
package main

import "fmt"

func main() {
	result := bubbleSort([]float64{1, 5, 4, 6})
	fmt.Println(result)
}

func bubbleSort(arr []float64) []float64 {
	n := len(arr)
	// in remain, find the biggest one to put in the last index
	for pass:=n-1;pass>=0;pass--{
		// rise up the big one in arr
		for i:=0;i<pass;i++{
			if arr[i]>arr[i+1]{
				temp := arr[i]
				arr[i] = arr[i+1]
				arr[i+1] = temp	
			}
		}
	}
	return arr
}

File: sorting_algorithms/bubble-sort/bubbleSort.py
def bubble_sort(arr):
    n = len(arr)
    # find ith biggest
    for i in range(n-1,-1,-1):
        # only need to compare i-1 times
        for j in range(i):
            if arr[j]>arr[j+1]:
                temp = arr[j]
                arr[j]=arr[j+1]
                arr[j+1]=temp
            
    return arr


if __name__=="__main__":
    arr = [3,5,1]
    arr = bubble_sort(arr)
    print(arr)